irohad/ordering/impl/ordering_service_impl.cpp

```diff
@@ -103,6 +118,8 @@ namespace iroha {
     }
 
     OrderingServiceImpl::~OrderingServiceImpl() {
+      std::lock_guard<std::mutex> lock(m);
+      is_finished = true;
```

Comment:
What's the purpose of setting variable value of the object that is being deleted?

Reply:
Mutex locks updateTimer and dtor. `is_finished` necessary for skipping the first one if dtor is already called, e.g:

call updateTimer & call dtor
dtor.lock()
dtor.unlock()
updateTimer.lock() <- after we just need to leave